.github/workflows/publish-crate.yml

```diff
@@ -0,0 +1,45 @@
+## This Source Code Form is subject to the terms of the Mozilla Public
+## License, v. 2.0. If a copy of the MPL was not distributed with this
+## file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+name: Publish Crate Workflow
+
+on:
+  push:
+    tags:
+      - 'v[0-9]+.[0-9]+.[0-9]+'
+
+    # ?? Should this run on merge/push to master instead of main (guessing you are using main like develop)?
+    # branches:
+    #   - main
+
+    # ?? Can potentially enable this to control manually when to publish.
+    # workflow_dispatch:
+
+  # TEMP: Turn this off after test
+  pull_request:
+    branches:
+      - main
+
+
+jobs:
+  publish-crate:
+    name: Publish crate job
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code into the directory
+        uses: actions/checkout@v3
+
+      - name: Install minimal and stable toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: stable
+          override: true
+
+      - name: Publish to crate.io using token
+        env:
+          CRATES_TOKEN: ${{ secrets.SOURCE_DEVS_ALL_PERMS_TOKEN_CRATES_IO }}
+        run: cargo publish --token ${CRATES_TOKEN}
```

Comment:
question:  This looks very insecure, is it?

Reply:
Is pretty standard practice: https://docs.github.com/en/actions/security-guides/using-secrets-in-github-actions#example-using-bash

Here is how the env var will show up in logs:
```
  env:
    CRATES_TOKEN: ***
```

However I noticed with cargo.io another practice can be to just set the environment var `CARGO_REGISTRY_TOKEN` with the secret which can pick it up. I can try that if you prefer it?

Reply:
Ah okay nice, and thanks for the explanation :)  I guess github must do some magic here then - no need to change, happy as is :)

Reply:
I did end up changing it, it does seem cleaner.